### PR TITLE
Task relocation unit and integration tests

### PR DESCRIPTION
**Pull request by tbak**
### Description of the Change

+ logging, metrics and bug fixes

### CODE REVIEW

File: titus-supplementary-component/task-relocation/src/main/java/com/netflix/titus/supplementary/relocation/endpoint/grpc/TaskRelocationGrpcService.java [hunk 1]
```diff
@@ -54,14 +67,42 @@ public void getCurrentTaskRelocationPlans(TaskRelocationQuery request, StreamObs
...
         responseObserver.onNext(grpcResults);
         responseObserver.onCompleted();
     }
 
+    @Override
+    public void getTaskRelocationResult(RelocationTaskId request, StreamObserver<TaskRelocationExecution> responseObserver) {
+        String taskId = request.getId();
+
+        TaskRelocationStatus latest = relocationWorkflowExecutor.getLastEvictionResults().get(taskId);
+        List<TaskRelocationStatus> archived = archiveStore.getTaskRelocationStatusList(taskId).block();
```
Comment (corindwyer) [inline titus-supplementary-component/task-relocation/src/main/java/com/netflix/titus/supplementary/relocation/endpoint/grpc/TaskRelocationGrpcService.java]: Don't block the grpc thread here.

File: titus-supplementary-component/task-relocation/src/main/java/com/netflix/titus/supplementary/relocation/descheduler/DefaultDeschedulerService.java [hunk 2]
```diff
@@ -88,16 +91,56 @@ public DefaultDeschedulerService(ReadOnlyJobOperations jobOperations,
...
+                                .withFailure(failure)
+                                .build()
+                );
+            }
+        }
+
+        return new ArrayList<>(result.values());
+    }
+
+    private boolean isMangedByTaskRelocationService(Task task) {
```
[the remark this answers is not in the document]
Comment (tbak) [reply titus-supplementary-component/task-relocation/src/main/java/com/netflix/titus/supplementary/relocation/descheduler/DefaultDeschedulerService.java]: Fixed

File: titus-supplementary-component/task-relocation/src/main/java/com/netflix/titus/supplementary/relocation/workflow/DeschedulingResultLogger.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+                logger.info("        Agent({}):", agentId);
+                results.forEach(result ->
+                        logger.info("           task({}): {}", result.getTask().getId(), doFormat(result.getTaskRelocationPlan()))
+                );
+            });
+        }
+        if (failureCount > 0) {
+            logger.info("    Not evictable tasks (failures):");
+            byAgentNotEvictable.forEach((agentId, results) -> {
+                logger.info("        Agent(%s):", agentId);
```
[the remark this answers is not in the document]
Comment (tbak) [reply titus-supplementary-component/task-relocation/src/main/java/com/netflix/titus/supplementary/relocation/workflow/DeschedulingResultLogger.java]: Fixed

File: titus-supplementary-component/task-relocation/src/test/java/com/netflix/titus/supplementary/relocation/workflow/step/MustBeRelocatedTaskCollectorStepTest.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        Job<BatchJobExt> job = TestDataFactory.newBatchJob("job1", 1, newSelfManagedDisruptionBudget(1_000));
+        relocationConnectorStubs.addJob(job);
+        relocationConnectorStubs.place(TestDataFactory.REMOVABLE_INSTANCE_GROUP, jobOperations.getTasks().get(0));
+
+        Map<String, TaskRelocationPlan> result = step.collectTasksThatMustBeRelocated();
+        assertThat(result).hasSize(1);
+    }
+
+    @Test
+    public void testTaskWithNoDisruptionBudgetAreNotMigrated() {
```
[the remark this answers is not in the document]
Comment (tbak) [reply titus-supplementary-component/task-relocation/src/test/java/com/netflix/titus/supplementary/relocation/workflow/step/MustBeRelocatedTaskCollectorStepTest.java]: Fixed

File: titus-supplementary-component/task-relocation/src/main/java/com/netflix/titus/supplementary/relocation/endpoint/grpc/TaskRelocationGrpcService.java (hunk 1 above)
Comment (tbak) [reply titus-supplementary-component/task-relocation/src/main/java/com/netflix/titus/supplementary/relocation/endpoint/grpc/TaskRelocationGrpcService.java]: Fixed

**Pull request merged**